src/zarr/testing/stateful.py

```diff
@@ -68,6 +68,9 @@ def can_add(self, path: str) -> bool:
     # -------------------- store operations -----------------------
     @rule(name=node_names, data=st.data())
     def add_group(self, name: str, data: DataObject) -> None:
+        # Handle possible case-insensitive file systems (e.g. MacOS)
+        if isinstance(self.store, LocalStore):
+            name = name.lower()
```

Comment:
This line doesn't seem to be covered by the tests, but looks like it's intended to be run? Do you know whether this is a coverage upload issue, or if the line actually isn't being run in tests?

Reply:
that's a bad report, or perhaps out-of-date. this PR also enables coverage recording for the hypothesis tests.